### PR TITLE
feat(kemptystate): redesign [khcp-1990]

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Update `KEmptyState` design.

#### Changes made:
* Add `icon` prop for customizing the displayed icon
* Add theming support for `--KEmptyTitleColor`, `--KEmptyContentColor`, `--KEmptyBackground`
* Update tests

Figma: https://www.figma.com/file/sfSZhMfpzCzddEhd8tdk9s/Kong_Design-library?node-id=257%3A1640

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: packages/KEmptyState/KEmptyState.vue [hunk 1]
```diff
@@ -2,13 +2,13 @@
   <div class="empty-state-wrapper">
```
Comment (chrisvdp) [inline packages/KEmptyState/KEmptyState.vue:2]: Probably a good usage of `<section>`

**Review by chrisvdp — CHANGES_REQUESTED** [review]

File: packages/KEmptyState/KEmptyState.vue [hunk 2]
```diff
@@ -65,22 +68,22 @@ export default {
...
-  color: rgba(0,0,0,.70);
-  border: 1px solid rgba(151,151,151,.1);
   border-radius: 3px;
-  background-color: rgba(0,0,0,.03);
+  background-color: var(--KEmptyBackground, var(--white));
 }
 .empty-state-wrapper h5 {
-  margin: 0 0 .75rem;
+  color: var(--KEmptyTitleColor, var(--black-500));
+  margin: 0 0 14px;
```
Comment (aanchalm01) [inline packages/KEmptyState/KEmptyState.vue:79]: Any specific reason for why we are switching from rem to px?

File: packages/KEmptyState/KEmptyState.vue (hunk 1 above)
Comment (kaiarrowood) [reply packages/KEmptyState/KEmptyState.vue]: Fixed

File: packages/KEmptyState/KEmptyState.vue (hunk 2 above)
Comment (kaiarrowood) [reply packages/KEmptyState/KEmptyState.vue:79]: This is a good question. For awhile `rem` was necessary due to `px` not scaling well, but now most browsers use zoom to increase scale so rem is actually no longer necessary. The first answer here has a pretty good explanation: https://stackoverflow.com/questions/11799236/should-i-use-px-or-rem-value-units-in-my-css

**Review by chrisvdp — APPROVED** [review]

**Pull request merged**